File: src/previous/day47/BinarySearchList.ts
export default function bs_list(haystack: number[], needle: number): boolean {
  let [lo, hi] = [0, haystack.length]

  while (lo < hi) {
    const mid = Math.floor(lo + (hi - lo) / 2)
    const val = haystack[mid]

    if (val === needle) return true
    if (needle < val) {
      hi = mid
    } else if (val < needle) {
      lo = mid + 1
    }
  }

  return false
}

File: go/day02/BinarySearchList/BinarySearchList.go
package binary_search_list

func BinarySearchList(haystack []int, needle int) bool {
	lo := 0
	hi := len(haystack)

	for lo < hi {
		mid := lo + ((hi - lo) / 2)
		val := haystack[mid]

		if needle == val {
			return true
		}

		if needle < val {
			hi = mid
		} else {
			lo = mid + 1
		}
	}

	return false
}
